src/conn/conn_open.c

```diff
@@ -205,6 +205,15 @@ __wt_connection_workers(WT_SESSION_IMPL *session, const char *cfg[])
     WT_RET(__wt_statlog_create(session, cfg));
     WT_RET(__wt_logmgr_create(session, cfg));
 
+    /* Initialize metadata tracking, required before creating tables. */
+    WT_RET(__wt_meta_track_init(session));
+
+    /*
+     * Try to create the history store table. Failure here suggests that the history file is
+     * corrupted in some way. We catch this failure and then run recovery.
+     */
+    WT_RET_ERROR_OK(__wt_hs_create(session, cfg), ENOENT);
+
     /*
      * Run recovery. NOTE: This call will start (and stop) eviction if recovery is required.
      * Recovery must run before the history store table is created (because recovery will update the
```

Comment:
This change directly contradicts the comment at this line. How do we resolve the issue in the comment? In any case this comment needs to change if creating the history store is now okay before recovery.